Advanced/Play Asset Delivery/Assets/PlayAssetDelivery/Editor/PlayAssetDeliveryBuildProcessor.cs

```diff
@@ -10,6 +10,108 @@
 {
+#if UNITY_2021_2_OR_NEWER
+    /// <summary>
+    /// Moves custom asset pack data from their default build location <see cref="BuildScriptPlayAssetDelivery"/> to their correct player build data location.
+    /// For an Android App Bundle, bundles assigned to a custom asset pack must be located in their {asset pack name}.androidpack directory in the Assets folder.
+    /// The 'CustomAssetPacksData.json' file is also added to the built player's StreamingAssets file location.
+    ///
+    /// This script executes before the <see cref="AddressablesPlayerBuildProcessor"/> which moves all Addressables data to StreamingAssets.
+    public class PlayAssetDeliveryBuildProcessor : BuildPlayerProcessor
+    {
+        /// <summary>
+        /// Returns the player build processor callback order.
+        /// </summary>
+        public override int callbackOrder
+        {
+            get { return 0; }
+        }
+
+        /// <summary>
+        /// Invoked before performing a Player build. Moves AssetBundles to their correct data location based on the build target platform.
+        /// </summary>
+        /// <param name="buildPlayerContext">Contains data related to the player.</param>
+        public override void PrepareForBuild(BuildPlayerContext buildPlayerContext)
+        {
+           if (EditorUserBuildSettings.activeBuildTarget == BuildTarget.Android && EditorUserBuildSettings.buildAppBundle)
+                MoveDataForAppBundleBuild(buildPlayerContext);
+            else
+                MoveDataToDefaultLocation();
+        }
+
+        /// <summary>
+        /// Move custom asset pack data from their build location to their App Bundle data location.
+        /// </summary>
+        /// <param name="buildPlayerContext">Contains data related to the player.</param>
+        public static void MoveDataForAppBundleBuild(BuildPlayerContext buildPlayerContext)
+        {
+            try
+            {
+                buildPlayerContext.AddAdditionalPathToStreamingAssets(CustomAssetPackUtility.CustomAssetPacksDataEditorPath, CustomAssetPackUtility.kCustomAssetPackDataFilename);
+                
+                AssetDatabase.StartAssetEditing();
+                if (File.Exists(CustomAssetPackUtility.BuildProcessorDataPath))
+                {
+                    string contents = File.ReadAllText(CustomAssetPackUtility.BuildProcessorDataPath);
+                    var data =  JsonUtility.FromJson<BuildProcessorData>(contents);
+
+                    foreach (BuildProcessorDataEntry entry in data.Entries)
+                    {
+                        string assetsFolderPath = Path.Combine(CustomAssetPackUtility.PackContentRootDirectory, entry.AssetsSubfolderPath);
+                        if (File.Exists(entry.BundleBuildPath))
+                        {
+                            File.Move(entry.BundleBuildPath, assetsFolderPath);
+                            File.Delete(entry.BundleBuildPath + ".meta");
```

Comment:
I think this should work fine. I know there has been talk about moving the .meta for folders, due to version control issues. But I think it was dropped due to potential breaking changes. Pops up from time to time. It might be safer to use https://docs.unity3d.com/ScriptReference/AssetDatabase.GetTextMetaFilePathFromAssetPath.html in case the behaviour changes though